File: backend/src/main/java/com/rental/modules/user/domain/entity/UserPreference.java
package com.rental.modules.user.domain.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "user_preferences")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserPreference {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "preference_id")
    private Long preferenceId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "min_budget")
    private Double minBudget;

    @Column(name = "max_budget")
    private Double maxBudget;

    @Column(name = "has_pet")
    private Boolean hasPet;

    @Column(name = "preferred_area", length = 150)
    private String preferredArea;
}

File: frontend/lib/features/preferences/data/models/user_preference.dart
class UserPreference {
  final double? minBudget;
  final double? maxBudget;
  final bool? hasPet;
  final String? preferredArea;

  const UserPreference({
    this.minBudget,
    this.maxBudget,
    this.hasPet,
    this.preferredArea,
  });

  factory UserPreference.fromJson(Map<String, dynamic> json) {
    return UserPreference(
      minBudget: (json['minBudget'] as num?)?.toDouble(),
      maxBudget: (json['maxBudget'] as num?)?.toDouble(),
      hasPet: json['hasPet'] as bool?,
      preferredArea: json['preferredArea'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'minBudget': minBudget,
        'maxBudget': maxBudget,
        'hasPet': hasPet,
        'preferredArea': preferredArea,
      };
}
